From e0a89768b36904ffc39ffde099b1c0d4de17938f Mon Sep 17 00:00:00 2001
From: Eric Bentley <eric.bentley@lairdtech.com>
Date: Wed, 29 Jan 2014 18:58:00 -0500
Subject: [PATCH] Allow connect to AP with multiple ciphers

The WB45 was not connecting to an AP configured for WPA2 but allowing
both AES and TKIP encryption.  When the WB45 was configured for wpa2-aes,
it would not connect to this mixed mode AP configuration.  It should
connect using WPA2 with AES as the unicast cipher and TKIP for the group
cipher but the WB45 would not attempt to connect (probes sent and probe responses, but never attempted to associate).

By or'ing in the ignore wpax cipher flag resolves the issue.

BZ5014
---
 drivers/net/wireless/ath/ath6kl/cfg80211.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath6kl/cfg80211.c b/drivers/net/wireless/ath/ath6kl/cfg80211.c
index 7822f25..1e8b968 100644
--- a/drivers/net/wireless/ath/ath6kl/cfg80211.c
+++ b/drivers/net/wireless/ath/ath6kl/cfg80211.c
@@ -614,7 +614,8 @@ static int ath6kl_cfg80211_connect(struct wiphy *wiphy, struct net_device *dev,
 					vif->grp_crypto, vif->grp_crypto_len,
 					vif->ssid_len, vif->ssid,
 					vif->req_bssid, vif->ch_hint,
-					ar->connect_ctrl_flags, nw_subtype);
+					ar->connect_ctrl_flags | CONNECT_IGNORE_WPAx_GROUP_CIPHER, 
+					nw_subtype);
 
 	/* disable background scan if period is 0 */
 	if (sme->bg_scan_period == 0)
-- 
1.7.9.5

